Dockerfile

```diff
@@ -1,45 +1,86 @@
-FROM --platform=$BUILDPLATFORM ubuntu as build
+# syntax=docker/dockerfile:1
+
+FROM --platform=${BUILDPLATFORM} ubuntu:24.04 AS builder-base
+# Configure the shell to exit early if any command fails, or when referencing unset variables.
+# Additionally `-x` outputs each command run, this is helpful for troubleshooting failures.
+SHELL ["/bin/bash", "-eux", "-o", "pipefail", "-c"]
+
+RUN \
+  --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+  --mount=target=/var/cache/apt,type=cache,sharing=locked \
+  <<HEREDOC
+    # https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#example-cache-apt-packages
+    # https://stackoverflow.com/questions/66808788/docker-can-you-cache-apt-get-package-installs#comment135104889_72851168
+    rm -f /etc/apt/apt.conf.d/docker-clean
+    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+
+    apt update && apt install -y --no-install-recommends \
+      build-essential \
+      curl \
+      python3-venv \
+      cmake
+HEREDOC
```

Comment:
The cache mount here (_and workaround for `docker-clean` in this base image used_), doesn't bring any savings to disk usage of this layer. It does however keep the cache external instead of the automatic clean, which is useful when the layer is invalidated, as it speeds up the build time.